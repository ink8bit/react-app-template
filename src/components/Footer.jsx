import React, { Component } from 'react';
import styled from '@emotion/styled';

import color from './../styles/color';
import offset from './../styles/offset';

const FooterContainer = styled.footer`
  background-color: ${color.FOOTER_BG};
  display: flex;
  padding: 10px ${offset.PADDING_COMMON};
  justify-content: center;
  align-items: center;
  height: ${offset.FOOTER_HEIGHT};

  li {
    display: inline-block;
  }

  li + li {
    margin-left: 20px;
  }
`;

class Footer extends Component {
  render() {
    return (
      <FooterContainer className="footer">
        <ul className="footer__links">
          <li className="footer__link">footer link 1</li>
          <li className="footer__link">footer link 2</li>
          <li className="footer__link">footer link 3</li>
        </ul>
      </FooterContainer>
    );
  }
}

export default Footer;
