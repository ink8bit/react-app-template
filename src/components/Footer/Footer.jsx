import React, { Component } from 'react';
import { Link } from '@reach/router';

import s from './styles.css';

class Footer extends Component {
  render() {
    return (
      <footer className={s.footer}>
        <ul>
          <li className={s.footerItem}>
            <Link className={s.footerLink} to="/">
              footer link 1
            </Link>
          </li>
          <li className={s.footerItem}>
            <Link className={s.footerLink} to="/">
              footer link 2
            </Link>
          </li>
          <li className={s.footerItem}>
            <Link className={s.footerLink} to="/">
              footer link 3
            </Link>
          </li>
        </ul>
      </footer>
    );
  }
}

export default Footer;
