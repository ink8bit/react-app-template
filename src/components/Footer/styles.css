.footer {
  --footer-height: 100px;

  background-color: #0c0c0c;
  display: flex;
  padding: 10px var(--padding);
  justify-content: center;
  align-items: center;
  height: var(--footer-height);
}

.footerItem {
  display: inline-block;
}

.footerItem + .footerItem {
  margin-left: 20px;
}

.footerLink {
  text-decoration: none;
  color: var(--link-color);
}

.footerLink:hover {
  text-decoration: underline;
}
