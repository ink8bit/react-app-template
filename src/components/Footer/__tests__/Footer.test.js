import React from 'react';
import { create } from 'react-test-renderer';

import Footer from './../Footer.jsx';

test('snapshot', () => {
  const c = create(<Footer />);
  expect(c.toJSON()).toMatchSnapshot();
});
