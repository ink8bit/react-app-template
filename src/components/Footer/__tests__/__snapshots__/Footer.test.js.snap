// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<footer
  className="footer css-x5faea"
>
  <ul
    className="footer__links"
  >
    <li
      className="footer__link"
    >
      <a
        aria-current="page"
        className="css-1bdxinm"
        href="/"
        onClick={[Function]}
      >
        footer link 1
      </a>
    </li>
    <li
      className="footer__link"
    >
      <a
        aria-current="page"
        className="css-1bdxinm"
        href="/"
        onClick={[Function]}
      >
        footer link 2
      </a>
    </li>
    <li
      className="footer__link"
    >
      <a
        aria-current="page"
        className="css-1bdxinm"
        href="/"
        onClick={[Function]}
      >
        footer link 3
      </a>
    </li>
  </ul>
</footer>
`;
